mkreportindex.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import collections
+import os
+import base64
+import sys
+sys.path.append('/afs/cern.ch/user/d/daho/SAS-build/scripts')
+import common
+import checkformat
+
+def get_diff_map(file_name):
+    '''
+    generates the diffmap for a single file
+    '''
+    clang_format_output, num_violations = common.CheckFormattingRules(file_name)
+    diff_lines = common._ComparisonReport(file_name, clang_format_output)
+    if num_violations != 0:
```

Comment:
since you already check whether you have any violations here, you can move the call to `ComparisonReport` in there as well.